src/udev/udevadm-cat.c

```diff
@@ -0,0 +1,137 @@
+               program_invocation_short_name,
+               ansi_highlight(),
+               ansi_normal(),
+               link);
+
+        return 0;
+}
+
+static int parse_argv(int argc, char *argv[]) {
+        enum {
+                ARG_ROOT = 0x100,
+                ARG_TLDR,
+                ARG_CONFIG,
+        };
+        static const struct option options[] = {
+                { "help",          no_argument,       NULL, 'h'             },
+                { "version",       no_argument,       NULL, 'V'             },
+                { "root",          required_argument, NULL, ARG_ROOT        },
+                { "tldr",          no_argument,       NULL, ARG_TLDR        },
+                { "config",        no_argument,       NULL, ARG_CONFIG      },
+                {}
+        };
+
+        int r, c;
+
+        assert(argc >= 0);
+        assert(argv);
+
+        while ((c = getopt_long(argc, argv, "hVN:", options, NULL)) >= 0)
+                switch (c) {
+                case 'h':
+                        return help();
+                case 'V':
+                        return print_version();
+                case ARG_ROOT:
+                        r = parse_path_argument(optarg, /* suppress_root= */ true, &arg_root);
+                        if (r < 0)
+                                return r;
+                        break;
+                case ARG_TLDR:
+                        arg_cat_flags = CAT_TLDR;
+                        break;
+                case ARG_CONFIG:
+                        arg_config = true;
+                        break;
+                case '?':
+                        return -EINVAL;
+                default:
+                        assert_not_reached();
+                }
+
+        if (arg_root && optind < argc)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Combination of --root= and FILEs is not supported.");
+
+        if (arg_config && optind < argc)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Combination of --config and FILEs is not supported.");
+
+        return 1;
+}
+
+int cat_main(int argc, char *argv[], void *userdata) {
+        int r;
+
+        r = parse_argv(argc, argv);
+        if (r <= 0)
+                return r;
+
+        if (arg_config)
+                return conf_files_cat(arg_root, "udev/udev.conf", arg_cat_flags);
+
+        _cleanup_strv_free_ char **files = NULL;
+        if (optind < argc) {
+                STRV_FOREACH(s, strv_skip(argv, optind)) {
+                        if (!endswith(*s, ".rules")) {
+                                log_warning("%s: File name must end with '.rules', ignoring.", *s);
+                                continue;
+                        }
+
+                        r = strv_extend(&files, *s);
+                        if (r < 0)
+                                return r;
+                }
```

Comment:
IIUC this means absolute path to udev rules is required? This seems at odds with systemctl/networkctl/machinectl cat?